### PR TITLE
Improve publish logic

### PR DESCRIPTION
**Pull request by Larusso**
## Description

This patch improves the `HockeyUploadTask` in two ways.

* switch to `apache:httpclient` for upload
* save upload result metadata `HockeyAppVersion`

I moved from using curl to `httpclient`. The setup is very simple if a little bit verbose. But it supports everything we need to `POST` a multipart payload. The main reason why I wanted to switch is to read the response body of the request. Hockey returns a JSON body with information about the created app version

**example from [support.hockeyapp.net][api-versions]
```json
{
    "title": "HockeyTest",
    "appsize": 1598428,
    "timestamp": 1308930206,
    "device_family": "iPhone/iPod",
    "minimum_os_version": "4.0",
    "notes": "<p>Some new features and fixed bugs.</p>",
    "version": "8",
    "shortversion": "1.0",
    "status": 2,
    "config_url": "https://rink.hockeyapp.net/manage/apps/123/app_versions/8",
    "public_url": "https://rink.hockeyapp.net/apps/1234567890abcdef1234567890abcdef"
}
```

> The actual body contains more fields.

The `HockeyUploadTask` will save this result as an `Outputfile` and also parse it into a simple model `HockeyAppVersion`. This allows other tasks to access this information.

## Changes

![ADD] new compile dependency `org.apache.httpcomponents:httpclient`
![ADD] new compile dependency `org.apache.httpcomponents:httpmime`
![IMPROVE] switch upload logic from `curl` to `httpclient`
![IMPROVE] save upload result body in file
![IMPROVE] provide runtime model for upload result

[api-versions]:    https://support.hockeyapp.net/kb/api/api-versions#upload-version

[NEW]:https://atlas-resources.wooga.com/icons/icon_new.svg "New"
[ADD]:http://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:http://resources.atlas.wooga.com/icons/icon_improve.svg "IMPROVE"
[CHANGE]:http://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:http://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:http://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:http://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:http://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:http://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:http://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:http://resources.atlas.wooga.com/icons/icon_webGL.svg "Web:GL"

### CODE REVIEW

File: src/main/groovy/wooga/gradle/hockey/tasks/HockeyUploadTask.groovy [hunk 1]
```diff
@@ -91,17 +113,33 @@ class HockeyUploadTask extends ConventionTask {
...
+                .addPart("ipa", binary)
+                .addTextBody("status", "2")
+                .addTextBody("notify", "2")
+                .build()
+
+        post.setEntity(content)
+        HttpResponse response = client.execute(post)
+
+        if(response.statusLine.statusCode != 201) {
+            throw new GradleException("unable to upload to hockey")
```
Comment (marcolink) [inline src/main/groovy/wooga/gradle/hockey/tasks/HockeyUploadTask.groovy:134]: could be more specific, maybe add the returned status code to the error message? or will it be exposed at some other point?

Comment (Larusso) [reply src/main/groovy/wooga/gradle/hockey/tasks/HockeyUploadTask.groovy:134]: At this point I only wanted to move from `curl` to `httpclient`. Curl will fail with exit code > 0 for anything else than a result in the `200` area. And the actual message we would see in gradle was/is limited. What I mean is: `Error handling and proper logging` was not my focus here. I just want to fail, when the result is not a `201` which is the documented success return from Hockey, to have the same basic behavior has the `curl` solution.

**Review by alexandershestakov — APPROVED** [review]
Awesome. And it works on Windows, right?

**Comment by Larusso** [comment]
@alexandershestakov hold my 🍺

**Comment by Larusso** [comment]
@alexandershestakov Yes it works!

**Comment by alexandershestakov** [comment]
Thanks for checking! Also really cool how it's even less than one line to check ;)

**Pull request merged**